engine/src/main/java/org/terasology/rendering/nui/contextMenu/ContextMenuTree.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.contextMenu;
+
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class ContextMenuTree {
```

Comment:
I personally would have gone for the name MenuTree, as it might be possible to have utility methods that create a menu tree that can be used both in context menu and in a menu at the top of a window.

Also a javadoc would be nice. that explains that what utlitiy class can be used to acutally show it as context menu.